### DISCUSSION TITLE
Can we still use `@NotificationMessage` with a `@SqsListener` annotated method? (`3.1.X`)

### DISCUSSION THREAD

**Discussion by kevinmcp123** [awspring/spring-cloud-aws#1035] [Q&A] [answered]
Hello 👋 I'm upgrading from `2.4.0` to `3.1.0` (SQS & SNS).

In version `2.4.0` I believe we could consume events from an SQS that is subscribed to an SNS via...

```
	
	@SqsListener(value = "my-queue")
	public void receiveEvent(
			@Valid @NonNull @NotificationMessage final Foo foo ) {
                ....
	}
```

I just wanted to confirm is this still the case in `3.1.0`?  I'm trying to do this and it tries to map `Foo` to the entire event meaning most of the fields end up as null & we fail validation (rather than only mapping `Foo` to the SNS `message` field).

The following _does work_ but ideally I'd like to keep parity with the previous solution. The SnsMessage approach also means the consumer is responsible for the JSON conversion of the `message` field where previously this was the framework.
```
	
	@SqsListener(value = "my-queue")
	public void receiveEvent(
			@Valid @NonNull final SnsMessage message ) {
                ....
	}
```

Thank you

**Comment by tomazfernandes** [comment, marked as answer]
Hi @kevinmcp123, unfortunately that feature is not in place yet, but should be in the coming weeks.

For now the alternative is manually mapping the Sns message to a POJO.

**Reply by kevinmcp123** [reply]
Thanks @tomazfernandes
